go/parquet/pqarrow/encode_arrow.go

```diff
@@ -246,6 +246,10 @@ type binary64arr interface {
 }
 
 func writeDenseArrow(ctx *arrowWriteContext, cw file.ColumnChunkWriter, leafArr arrow.Array, defLevels, repLevels []int16, maybeParentNulls bool) (err error) {
+	if leafArr.Len() == 0 {
```

Comment:
Can't comment on it directly since it's not in the diff, but it's not immediately obvious to me why we'd handle this case differently: https://github.com/apache/arrow/blob/c8363fa209fed4dfa30a0eee70283d62fc6f1627/go/parquet/pqarrow/encode_arrow.go#L270-L275

Also, it seems like that branch may never be taken now? Unless it's only there to cover an edge-case where `leafArr` was reassigned to an empty storage array after the initial `Len() == 0` check: https://github.com/apache/arrow/blob/c8363fa209fed4dfa30a0eee70283d62fc6f1627/go/parquet/pqarrow/encode_arrow.go#L253-L257

Reply:
Turns out you're right and it ended up causing a different test to fail. So it looks like i just needed to add the checks for whether or not the buffer is nil *sigh*. I've updated this